pkg/operator/ceph/object/spec.go

```diff
@@ -461,6 +468,20 @@ func (c *clusterConfig) makeDaemonContainer(rgwConfig *rgwConfig) (v1.Container,
 	return container, nil
 }
 
+func (c *clusterConfig) getRGWContainerCMD(args *[]string) []string {
+	if c.isLocalizedReadAffinitySet() {
+		// $hostName in the args comes from the bash script variable with the same name.
+		*args = append(*args, cephconfig.NewFlag("crush location", fmt.Sprintf("%s=%s", "host", "$hostName")))
+		parseArgsForScript(args)
```

Comment:
Can we just always put quotes around the arguments that need them while we are initially generating the args? Then we don't need to iterate over them here.